android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2495,21 +2499,19 @@ private static void calculateDefaultKeyboardHeights(Context context) {
 
   /**
    * Get the navigation bar height (if visible) in pixels
-   * @param context - the context
+   * @param context - The context
+   * @param {KeyboardType} - KeyboardType.KEYBOARD_TYPE_INAPP or KeyboardType.KEYBOARD_TYPE_SYSTEM
    * @return int - navigation bar height in pixels
    */
-  public static int getNavigationBarHeight(Context context) {
-    // Determine if navigation bar is visible
-    int resourceId = context.getResources().getIdentifier(
-      "config_showNavigationBar", "bool", "android");
-    if (resourceId <= 0) {
-      return 0;
-    }
-
-    // Navigation bar visible so get the height
-    resourceId = context.getResources().getIdentifier(
-      "navigation_bar_height", "dimen", "android");
-    return (resourceId > 0) ? context.getResources().getDimensionPixelSize(resourceId) : 0;
+  public static int getNavigationBarHeight(Context context, KeyboardType keyboardType) {
+    // bottomInset already updated from the inset listener
+    if (keyboardType == KeyboardType.KEYBOARD_TYPE_INAPP) {
+      return inappBottomInset;
+    } else if (keyboardType == KeyboardType.KEYBOARD_TYPE_SYSTEM) {
+      return sysBottomInset;
+    }
+
+    return inappBottomInset;
```

Comment:
Can we guarantee that the `inappBottomInset` and `sysBottomInset` values will be set when this is called? Is it possible to calculate them here if they have not already been set?

Reply:
I wonder if we fallback to the previous computation of the navigation bar height.
It's inconsistent between gesture vs 3-button navigation mode, but still maybe better than uninitialized...

https://github.com/keymanapp/keyman/blob/c5b3598ab6ae99ea51b9f2ee9d2e7d928f351ccc/android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java#L2509-L2512

Reply:
If we do that, can we log an event to Sentry, so that we can hopefully track if this is arising as a problem anywhere?